File: LiarBar-back/src/main/java/org/huex/liarbarback/models/Room.java
package org.huex.liarbarback.models;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;
import java.util.concurrent.CopyOnWriteArrayList;

import lombok.*;

@Getter @Setter @AllArgsConstructor
public class Room {
    private String id;
    private List<Player> playerList;
    private int maxPlayers;
    private boolean isStarted;

    public Room(String id) {
        this.id=id;
        playerList = new CopyOnWriteArrayList<>();
        maxPlayers = 8;
    }

    public Player getPlayer(String userId) {
        for (Player player : playerList) {
            if (player.getUserId().equals(userId)) {
                return player;
            }
        }
        return null; 
    }

    public void addPlayer(Player player) {
        playerList.add(player);
    }

    public boolean removePlayer(String userId) {
        Player player = getPlayer(userId);
        if (player != null) {
            if (player.isHost() && playerList.size() > 1) {
                playerList.get(1).setHost(true);
            }
            playerList.remove(player);
            return true;
        }
        return false;
    }

    public boolean isFull() {
        return playerList.size() >= maxPlayers;
    }

    @Override
    public String toString() {
        String str = "Room " +id + "\n"
            + "ID\tName\tStatus\tReady\tHost\n"
            + "--------------------------------------------\n";
        for (Player player : playerList) {
            str += player.toString();
        }
        str+="--------------------------------------------\n";
        return str;
    }




    
}

File: LiarBar_unity20220322f1/Assets/Scripts/Models/Room.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

[Serializable]
public class Room
{
    public string id;
    public List<Player> playerList;
    public int maxPlayers;
    public bool started;

    public Room(string id, int maxPlayers)
    {
        this.id = id;
        this.maxPlayers = maxPlayers;
        playerList = new List<Player>();
        started = false;
    }
    public Room() { }
    public override string ToString()
    {
        return JsonUtility.ToJson(this);
    }
}
